Core/Main/itkTransformixFilter.hxx

```diff
@@ -190,6 +196,77 @@ TransformixFilter<TMovingImage>::GenerateData()
     }
   }
 
+  if (m_Transform)
+  {
+    // Adjust the local transformParameterMap according to this m_Transform.
+
+    const auto transformToMap = [](const itk::TransformBase & transform, auto & transformParameterMap) {
+      const auto convertToParameterValues = [](const itk::OptimizerParameters<double> & optimizerParameters) {
+        ParameterValueVectorType parameterValues(optimizerParameters.size());
+        std::transform(optimizerParameters.begin(),
+                       optimizerParameters.end(),
+                       parameterValues.begin(),
+                       itk::NumberToString<double>{});
+        return parameterValues;
+      };
+
+      transformParameterMap["ITKTransformFixedParameters"] = convertToParameterValues(transform.GetFixedParameters());
+      transformParameterMap["ITKTransformParameters"] = convertToParameterValues(transform.GetParameters());
+      transformParameterMap["ITKTransformType"] = { transform.GetTransformTypeAsString() };
+      transformParameterMap["Transform"] = { elx::TransformIO::ConvertITKNameOfClassToElastixClassName(
+        transform.GetNameOfClass()) };
+    };
+    const auto compositeTransform =
+      dynamic_cast<const CompositeTransform<double, MovingImageDimension> *>(&*m_Transform);
+
+    if (compositeTransform)
+    {
+      const auto & transformQueue = compositeTransform->GetTransformQueue();
+
+      const auto numberOfTransforms = transformQueue.size();
+
+      if (numberOfTransforms == 0)
+      {
+        itkExceptionMacro(
+          "The specified composite transform has no subtransforms! At least one subtransform is required!");
+      }
+
+      if (numberOfTransforms != numberOfTransformParameterMaps)
+      {
+        if (numberOfTransformParameterMaps == 0)
+        {
+          transformParameterMapVector.resize(numberOfTransforms);
+        }
+        else
+        {
+          // The last TransformParameterMap is special, as it needs to be used for the final transformation.
+          const auto lastTransformParameterMap = transformParameterMapVector.back();
+          transformParameterMapVector.resize(numberOfTransforms);
+          transformParameterMapVector.back() = lastTransformParameterMap;
+        }
+      }
+      for (unsigned int i = 0; i < numberOfTransforms; ++i)
+      {
+        auto &     transformParameterMap = transformParameterMapVector[numberOfTransforms - i - 1];
+        const auto transform = transformQueue[i];
+
+        if (transform == nullptr)
+        {
+          itkExceptionMacro("One of the subtransforms of the specified composite transform is null!");
+        }
+        transformToMap(*transform, transformParameterMap);
+      }
+    }
+    else
+    {
+      for (auto & transformParameterMap : transformParameterMapVector)
+      {
+        // Use the same transform for all parameter maps.
+        transformToMap(*m_Transform, transformParameterMap);
+      }
```

Comment:
@mstaring @stefanklein Also please check if you like this: When the specified transform is a single transform (not a composite), its Parameters/FixedParameters/etc are copied to each map. Also if there are multiple maps in the vector. Is that OK?

Should there be a check if there is at least one map? Or if there is _exactly_ one, when the transform is also a single one.

Reply:
Discussed with Marius (@mstaring) at the office: when there is only one single transform, there should be only one parameter map. If the specified `TransformParameterObject` has more than one map, only the last map should be taken into account.